client/consensus/slots/src/lib.rs

```diff
@@ -242,6 +246,36 @@ pub trait SimpleSlotWorker<B: BlockT> {
 		Some(proposal)
 	}
 
+	/// Calls `create_inherent_data` and handles errors.
+	async fn create_inherent_data(
+		slot_info: &SlotInfo<B>,
+		logging_target: &str,
+	) -> Option<sp_inherents::InherentData> {
+		let inherent_data = match slot_info.create_inherent_data.create_inherent_data().await {
+			Ok(data) => data,
+			Err(err) => {
+				warn!(
+					target: logging_target,
+					"Unable to create inherent data for for block {:?}: {}",
+					slot_info.chain_head.hash(),
+					err,
+				);
+				return None
+			},
+		};
+
+		if Instant::now() > slot_info.ends_at {
+			warn!(
+				target: "slots",
+				"Creating inherent data took more time than we had left for the slot.",
+			);
```

Comment:
nit: Maybe also log the block info here?

Reply:
Good idea